wallets/terra-extension/src/extension/signer.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  OfflineDirectSigner,
+  AccountData,
+  Registry,
+  DirectSignResponse,
+  decodePubkey,
+} from '@cosmjs/proto-signing';
+import { StdSignature } from '@cosmjs/amino';
+import { WalletAccount } from '@cosmos-kit/core';
+import { TerraExtension } from './extension';
+import { defaultRegistryTypes as defaultStargateTypes } from '@cosmjs/stargate';
+import { wasmTypes } from '@cosmjs/cosmwasm-stargate/build/modules/index';
+import { TxBody, AuthInfo } from 'cosmjs-types/cosmos/tx/v1beta1/tx';
+import { camelToSnake, FakeMsg } from './utils';
+import { Fee } from '@terra-money/feather.js';
+
+export interface SignDoc {
+  bodyBytes: Uint8Array;
+  authInfoBytes: Uint8Array;
+  chainId: string;
+  accountNumber: Long;
+}
+
+export class OfflineSigner implements OfflineDirectSigner {
+  private client: TerraExtension;
+  accountInfo: WalletAccount;
+
+  constructor(client: TerraExtension, accountInfo: WalletAccount) {
+    this.client = client;
+    this.accountInfo = accountInfo;
+  }
+
+  async getAccounts(): Promise<readonly AccountData[]> {
+    return [
+      {
+        address: this.accountInfo.address,
+        algo: this.accountInfo.algo || 'secp256k1',
+        pubkey: this.accountInfo.pubkey,
+      },
+    ];
+  }
+
+  async signDirect(
+    _signerAddress: string,
+    _signDoc: SignDoc
+  ): Promise<DirectSignResponse> {
+    const typeUrls = TxBody.decode(_signDoc.bodyBytes).messages.map(
+      ({ typeUrl }) => typeUrl
+    );
+
+    const registry = new Registry([...defaultStargateTypes, ...wasmTypes]);
```

Comment:
How do we add new types to the registry? E.g. if want to add osmosis messages

Reply:
@javiersuweijie This should be it? https://github.com/cosmos/cosmjs/blob/main/packages/proto-signing/src/registry.ts#L77-L91

Reply:
The issue here is that we cannot use `signDirect` to sign custom module messages (like Osmosis or Stride messages) since the only registered types are default cosmos module and cosmwasm. IMO we should be using `signAmino` for custom messages anyway so that the wallet can display the data in a human readable fashion.

Reply:
I've already implemented `signAmino` function.